### DISCUSSION TITLE
Issue dumping Road Rash - Sega Genesis

### DISCUSSION THREAD

**Discussion by PolloLoco88** [sanni/cartreader#782] [Need help]
I'm trying to dump Road Rash for Sega Genesis but seem to be getting a different CRC with each attempt, none of which match the CRC for a known good dump of Road Rash (USA).   I've attempted dumping two different copies of the game to be absolutely sure and have also given both PCB pins a thorough cleaning with a pink eraser and contact cleaner.  Both copies of the game run fine on my original Genesis model 1 and the dumps do load on BlastEm emulator.  I've had no issues dumping Genesis/Mega Drive carts on the hardware to this point so I'm stumped, definitely open to any suggestions on what else to try or if I should supply any more information for troubleshooting.  OSCR log excerpt below and each attempt was done with a different copy of Road Rash:

OSCR HW3 V12.4

[+] Mega Drive/Genesis

[+] Game Cartridge

Cart Info
 
Name: RoadRash
Size: 6 MBit
ChkS: 1EF5
Sram: None
 
Press Button...

[+] Read Rom

Saving to MD/ROM/RoadRash/356/...
[*******************]
Internal checksum...Error
A2A4

CRC32... DB9E28E1 -> Not found
Press Button...

OSCR HW3 V12.4

[+] Mega Drive/Genesis

[+] Game Cartridge

Cart Info
 
Name: RoadRash
Size: 6 MBit
ChkS: 1EF5
Sram: None
 
Press Button...

[+] Read Rom

Saving to MD/ROM/RoadRash/357/...
[*******************]
Internal checksum...Error
16A4

CRC32... E85D9848 -> Not found
Press Button...

**Comment by PsyK0p4T** [comment]
Hi.
The cart detection started well at least, header checksum "1EF5" was the expected value.

Can you check two things : 
- Once completed, what is your rom size ? 786432 bytes ?
- Can you post some pictures of the PCBs into your two cartridges ? Because Road Rash had different PCB revisions. One "two-chips" produced around 91-92 (which I'm unsure if OSCR handles it fine), and another produced later in 93-94 with just one chip.

If you're familiar with hex editors, you can also try to compare the roms you generated with the spreaded one (crc32 = dea53d19), to find what's different between them.

**Reply by PolloLoco88** [reply]
Hello, thank you for the quick reply!  I checked the following as you've suggested with the following results:

- Yes, every dump I attempt produces a rom size of 786432 bytes
- Photos of the front and back of the PCBs are below, sure enough both versions appear to be "two-chip" revisions

I also did a hex compare of the two different PCB dumps to the known good dump as a control (CRC32 = dea53d19).  Unfortunately I don't have the knowledge to interpret the results but both comparisons have differences starting at address 00080000 with the last difference at address 000BFFF8; both show an identical amount of differences to the good dump with 32,768 lines.

![RR_PCB_Front - Copy (Large)](https://user-images.githubusercontent.com/132236988/235569770-25157445-404e-4eb8-9e1f-706974de80dc.jpg)
![RR_PCB_Rear - Copy (Large)](https://user-images.githubusercontent.com/132236988/235569786-e34284b5-351f-463e-83d1-7e702a28b927.jpg)

**Reply by PsyK0p4T** [reply]
Thanks for your answers and pictures ! So, about the size, we're good at least.
I've already dumped some Electronic Arts cartridges with OSCR without any problem, like "LHX Attack Chopper", but this one was on a single chip. Here I don't know if the mapping could be different due to multiple chips.
Is it possible for you to come on our Discord server ? I can give you a script to force a big overdump, and then try to figure out how/where data are located with your results. But no guarantee :/

**Reply by PolloLoco88** [reply]
Hi PsyK0p4T, apologies for the delay - I'll try and drop you a line on the OSCR discord as per your suggestion.  FWIW I was curious about that extra chip on the "2-chip" version of Road Rash labeled SN74LS138N.  I looked up the part number and it looks like a demultiplexer of some kind, maybe this is what's causing the issues with OSCR as the rom data has to be accessed through this hardware?

https://www.alldatasheet.com/datasheet-pdf/pdf/5654/MOTOROLA/SN74LS138N.html

**Comment by retrovik** [comment]
I am also not able to dump Road Rash for the Sega Genesis. I get a different CRC every time with the Cartreader. The pins are extremely clean. However, I noticed that if I use a Retrode to dump this game, I get a consistent CRC. That CRC matches the GoodGen entry:

<hash crc32="465e8650" sha1="77dbef0d7fb03f7eabb933b7cc281770b8f73aea" name="Road Rash (UE) [o2]">

Road Rash (UE) [o2].bin
CRC32: 465e8650
SHA1: 77dbef0d7fb03f7eabb933b7cc281770b8f73aea

The size is: 786,432 bytes. That is the correct size, so I do not understand how this would be flagged as an "overdump". I thought overdumps meant more data than required. This has the exact required size. 

Another strange issue is this game is in a non standard Genesis case. It almost looks like a rental case. However the insert is professionally printed and looks authentic (but it too big to fit a normal EA Genesis case). The person I bought it from 100% promised it was authentic and not a reproduction / aftermarket case. I also found one just like it on eBay. It makes me think there is something very odd about this particular copy of the game. 

I would love to be able to dump this and get the "known" good CRC, but I am more confused than ever.


Was any solution found to the original post? Thanks!

**Reply by PsyK0p4T** [reply]
The cart case (but also the PCB) are unusual because they were not manufactured by SEGA. Electronic Arts or Codemasters decided to go with their own hardware. Your game is fully legit.

For Road Rash, it seems that two revisions of the pcb were done. One version used two chips (maybe 512 + 256 kB ?) and a later one used a bigger single chip. 
I don't have the cart so I can't test, but maybe Retrode firmware handles this two chips pcb better than OSCR can currently.
According to Datomatic, 786432 is the real expected size, not an overdump.

**Reply by retrovik** [reply]
Thanks for the response. I would be happy to help you test if that could work. 

Based on your reply, would that indicates that maybe the "overdump" version the Retrode creates (which is the correct size), was simply not handled correctly by the GoodGen tool back in the day (and is not really an overdump)? Or maybe it means that some of the dumpers 25+ years ago, did not know how to dump the two chip version of the game (and maybe it is really a bad and not an overdump). Or maybe even this would be considered another revision of the game? Not sure what (if any) conclusions I could draw at this point.

In the end, I am really just trying to determine if my cart is valid for being 100% playable (or if I need to find another one for preservation sake in my personal collection).

Anyway, thanks for the info.

My board is even different than the one in the opening post. Two different chips and different board / pin configuration. FYI:

![IMG_6406](https://github.com/sanni/cartreader/assets/122476671/aa52d03c-f0b3-494a-8d4e-6bd34dc12b3d)

![IMG_6407](https://github.com/sanni/cartreader/assets/122476671/be76874f-be1b-43d5-9239-087e54d235b8)

**Comment by jd213** [comment]
My copy of Road Rash also gives an error:

```
OSCR HW5 V12.3

[+] Mega Drive/Genesis

[+] Game Cartridge


Cart Info
 
Name: RoadRash
Size: 6 MBit
ChkS: 1EF5
Sram: None
 
Press Button...

[+] Read Rom


Saving to MD/ROM/RoadRash/472/...
[*******************]
Internal checksum...Error
BAA4

CRC32... E2186B73 -> Not found
```
(haven't updated my V5 Sanni since the 12.3 release though, last time I tried to update it the Arduino software said the upload was too big or something, so I guess I need to figure out how to fix that)

So far the dump seems to work fine, was even able to use the ROAD_RASH_Improvement_v1.1 patch on it.

Here's what my board looks like, apparently it's Rev A, but it looks different from retrovik's board, which also says Rev A: 
https://i.postimg.cc/T2s9Z7sj/DSC04666.jpg
https://i.postimg.cc/ncQkZK8y/DSC04667.jpg
